### PR TITLE
Update codes based on prior code review for item #21

### PR DESCRIPTION
**Pull request by WhiteHsu**
@sebbrudzinski 
I've updated codes based on your code review for Item #21. Please kindly help review again. Thank you very much.  :-)

### CODE REVIEW

**Comment by WhiteHsu** [comment]
@sebbrudzinski 
I've removed unrequired commit and please help review. Thank you.  :-)

**Comment by WhiteHsu** [comment]
@sebbrudzinski 
Is there any problem in my updates? Please kindly help review my codes and provide your valuable feedback. Thank you very much.  :-)

**Comment by WhiteHsu** [comment]
May anybody help review my codes and provide your valuable feedback? I hope to finish this workitem and select the next to work. Thank you very much.  :-)

**Comment by WhiteHsu** [comment]
May anybody help review my codes and provide your valuable feedback? I hope to finish this workitem and select the next to work. Thank you very much.  :-)

**Comment by sebbrudzinski** [comment]
@WhiteHsu Sorry it takes so long. I'll try to have it reviewed as soon as possible.

**Comment by WhiteHsu** [comment]
@sebbrudzinski 
Thanks for reply. Please kindly let me if any suggestion or feedback. :-)

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/DbxAutoSync.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public void run() {
+        while(this.interval > 0 && Cloud.getInstance().getStatus() == Cloud.Status.DBX_CONNECTED) {
+            try {                                
+                Thread.sleep(this.interval * 1000 * 60);
+                
+                SwingUtilities.invokeLater(new Runnable(){
+                    @Override
+                    public void run(){
+                        new DbxFileActions().saveProgress(DbxUtil.getDbxClient(), drtc, false);
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/DbxAutoSync.java]: Why is this wrapped in SwingUtilities.invokeLater()?

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/DbxAutoSync.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    }
+                }); 
+                
+            } catch (InterruptedException ex) {
+                break;
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/DbxAutoSync.java]: We need to log something here.

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/remote/RemoteAutoSync.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                "Period of Dropbox Remote Sync",
+                JOptionPane.QUESTION_MESSAGE,
+                null,
+                periodOptions,
+                periodOptions[0]
+        );                                
+
+        LOGGER.log("You set your interval of Dropbox auto sync as: " + syncPeriod);
+        
+        if(syncPeriod == null) {
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/remote/RemoteAutoSync.java]: Seems like this should be above the log statement

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/remote/RemoteAutoSync.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    public void actionPerformed(ActionEvent e) {        
+        DbxState dbxState = etc.getEditorState().getDbxState();       
+        
+        if(Cloud.getInstance().getStatus() != Cloud.Status.DBX_CONNECTED || dbxState == null) {
+            JOptionPane.showMessageDialog(null, "No Dropbox file has been loaded.\n"
+                        + "You must open Dropbox file, before you save it.", "Cannot save progress", JOptionPane.WARNING_MESSAGE);
+           return; 
+        }
+        
+        String[] periodOptions = {"Disable", "1 minute", "3 minutes", "5 minutes", "10 minutes"};
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/remote/RemoteAutoSync.java]: Make it an enum

File: OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/remote/RemoteAutoSync.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        );                                
+
+        LOGGER.log("You set your interval of Dropbox auto sync as: " + syncPeriod);
+        
+        if(syncPeriod == null) {
+            return;             
+        }
+        
+        if(!syncPeriod.startsWith("Disable")) {
+            syncPeriod = syncPeriod.substring(0, syncPeriod.indexOf(" "));
```
Comment (sebbrudzinski) [inline OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/remote/RemoteAutoSync.java]: this should be retrieved from an integer enum field

**Comment by WhiteHsu** [comment]
@sebbrudzinski 
Thanks for your comments. I'll update codes accordingly. However, for the method "SwingUtilities.invokeLater()", because the method "saveProgress()" will call the Windows System API, if we dont wrap it with method "SwingUtilities.invokeLater()", there'll be exception thrown as below:
pic01

Here's the exception message:
[java.lang.IllegalStateException.txt](https://github.com/sebbrudzinski/Open-LaTeX-Studio/files/556310/java.lang.IllegalStateException.txt)

This's my first time to encounter this problem and I've tried to find out solution via internet. There're different webpages recommending this solution with "SwingUtilities.invokeLater()". Here's example:
http://dev.platform.netbeans.narkive.com/IT166Fcp/creating-topcomponent-from-thread
http://stackoverflow.com/questions/12464524/getting-java-lang-illegalstateexception-this-call-must-happen-in-the-awt-event

After much try and I find this really works, so I've developed codes as you see. Or do you have any other suggested way? Please help feedback and advise. Thank you.  :-)

**Comment by sebbrudzinski** [comment]
Thanks for letting me know. I've got no problem having it in the code, just wanted to know the reason we need it in the codebase.

**Comment by WhiteHsu** [comment]
@sebbrudzinski 
Thanks for your feedback. I've updated codes based on your comments. Please help give it a look. Thank you.  :-)

**Pull request merged**